VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -273,45 +274,42 @@ private void updateUIForOrientation(){
     }
 
     private class CourseUnitPagerAdapter extends FragmentStatePagerAdapter {
+        private boolean isRefresh = false;
 
         public CourseUnitPagerAdapter(FragmentManager fm) {
             super(fm);
         }
 
-
-        public CourseComponent getUnit(int pos){
-            if ( pos >= unitList.size() )
-                pos = unitList.size() -1;
-            if ( pos < 0 )
+        public CourseComponent getUnit(int pos) {
+            if (pos >= unitList.size())
+                pos = unitList.size() - 1;
+            if (pos < 0)
                 pos = 0;
             return unitList.get(pos);
         }
+
         @Override
         public Fragment getItem(int pos) {
             CourseComponent unit = getUnit(pos);
-            CourseUnitFragment unitFragment = null;
+            CourseUnitFragment unitFragment;
             //FIXME - for the video, let's ignore studentViewMultiDevice for now
-            if ( unit instanceof VideoBlockModel) {
-                CourseUnitVideoFragment fragment = CourseUnitVideoFragment.newInstance((VideoBlockModel)unit);
-                unitFragment = fragment;
-            }
-
-            else if ( !unit.isMultiDevice() ){
+            if (unit instanceof VideoBlockModel) {
+                unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+            } else if (!unit.isMultiDevice()) {
                 unitFragment = CourseUnitMobileNotSupportedFragment.newInstance(unit);
-            }
-
-            else if ( unit.getType() != BlockType.VIDEO &&
-                unit.getType() != BlockType.HTML &&
-                unit.getType() != BlockType.OTHERS &&
-                unit.getType() != BlockType.DISCUSSION &&
-                unit.getType() != BlockType.PROBLEM ) {
+            } else if (unit.getType() != BlockType.VIDEO &&
+                    unit.getType() != BlockType.HTML &&
+                    unit.getType() != BlockType.OTHERS &&
+                    unit.getType() != BlockType.DISCUSSION &&
+                    unit.getType() != BlockType.PROBLEM) {
                 unitFragment = CourseUnitEmptyFragment.newInstance(unit);
+            } else if (unit instanceof HtmlBlockModel) {
+                if (!NetworkUtil.isConnected(CourseUnitNavigationActivity.this)) {
+                    unitFragment = CourseUnitNotAvailableFragment.newInstance(unit);
```

Comment:
Instead of creating and displaying a separate fragment in the case of non-connectivity, the error message should be displayed by `CourseUnitWebviewFragment` itself. This is all the more needed because we're also performing automatic reloading upon connectivity restoration, which is the task of the fragment and is out of scope here. In order to receive connectivity events on the Fragment, you could implement `NetworkObserver` and register for the events from the Activity, as done by `CourseListTabFragment`, or you could just subscribe to the events directly (which is really the pattern to use; `NetworkObserver` is an anti-pattern and should be ultimately removed).